llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -4153,22 +4153,23 @@ SDValue AMDGPUTargetLowering::performSraCombine(SDNode *N,
   SDLoc SL(N);
   unsigned RHSVal = RHS->getZExtValue();
 
-  // (sra i64:x, 32) -> build_pair x, (sra hi_32(x), 31)
-  if (RHSVal == 32) {
+  // For C >= 32
+  // (sra i64:x, C) -> build_pair (sra hi_32(x), C - 32), (sra hi_32(x), 31)
```

Comment:
This fails alive: https://alive2.llvm.org/ce/z/ohLB4J

We need a freeze somewhere, maybe this https://alive2.llvm.org/ce/z/kcwGVF (but need to scale it down to a smaller size to get this to complete)

Reply:
Can you post the scaled down alive link that shows this works